controllers/usernamespace/usernamespace_controller.go

```diff
@@ -310,57 +310,16 @@ func (r *CheUserNamespaceReconciler) reconcileTrustedCerts(ctx context.Context,
 	if err := deleteLegacyObject("trusted-ca-certs", &corev1.ConfigMap{}, targetNs, checluster, deployContext); err != nil {
 		return err
 	}
-	targetConfigMapName := prefixedName("trusted-ca-certs")
 
-	delConfigMap := func() error {
-		_, err := deploy.Delete(deployContext, client.ObjectKey{Name: targetConfigMapName, Namespace: targetNs}, &corev1.ConfigMap{})
-		return err
-	}
-
-	sourceMap := &corev1.ConfigMap{}
-	if err := r.client.Get(ctx, client.ObjectKey{Name: tls.CheMergedCABundleCertsCMName, Namespace: checluster.Namespace}, sourceMap); err != nil {
-		if !errors.IsNotFound(err) {
-			return err
-		}
-
-		return delConfigMap()
-	}
-
-	targetMap := &corev1.ConfigMap{
-		TypeMeta: metav1.TypeMeta{
-			Kind:       "ConfigMap",
-			APIVersion: "v1",
-		},
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      targetConfigMapName,
-			Namespace: targetNs,
-			Labels: defaults.AddStandardLabelsForComponent(checluster, userSettingsComponentLabelValue, map[string]string{
-				dwconstants.DevWorkspaceMountLabel:          "true",
-				dwconstants.DevWorkspaceWatchConfigMapLabel: "true",
-			}),
-			Annotations: addToFirst(sourceMap.Annotations, map[string]string{
-				dwconstants.DevWorkspaceMountAsAnnotation:   "file",
-				dwconstants.DevWorkspaceMountPathAnnotation: "/public-certs",
-			}),
-		},
-		Data: sourceMap.Data,
-	}
+	// Remove trusted-ca-certs ConfigMap from the target namespace to reduce the number of ConfigMaps
+	// and avoid mounting the same certificates under different paths.
+	// See cerificates#syncCheCABundleCerts
+	trustedCACertsCMKey := client.ObjectKey{Name: prefixedName("trusted-ca-certs"), Namespace: targetNs}
```

Comment:
on dogfooding, I see there is `che-trusted-ca-certs`, is it going to be removed by prefix?

Reply:
Also, testing configuration of 3.19, am I correct that `che-trusted-ca-certs` and `ca-certs-merged` are pretty much identical?

<img width="1587" alt="Screenshot 2025-03-19 at 14 21 37" src="https://github.com/user-attachments/assets/8c7b750f-32fa-4b51-9abf-686f7e3f6d9f" />

Reply:
1. che-trusted-ca-certs will be removed completedly
2. yes, those CMs are identical